src/datanode/src/store.rs

```diff
@@ -171,3 +175,12 @@ pub(crate) fn build_http_client() -> Result<HttpClient> {
 
     HttpClient::build(http_builder).context(error::InitBackendSnafu)
 }
+
+struct PrintDetailedError;
+
+// PrintDetailedError is a retry interceptor that prints error in Debug format in retrying.
+impl RetryInterceptor for PrintDetailedError {
+    fn intercept(&self, err: &Error, dur: Duration) {
+        warn!("Retry after {}s, error: {:?}", dur.as_secs_f64(), err);
```

Comment:
https://docs.rs/opendal/latest/opendal/struct.Error.html#display
Maybe we should use alternate format to avoid printing the backtrace
```suggestion
        warn!("Retry after {}s, error: {:#?}", dur.as_secs_f64(), err);
```